tmail-backend/combined-identity/src/main/java/com/linagora/tmail/combined/identity/CombinedUsersRepository.java

```diff
@@ -0,0 +1,87 @@
+package com.linagora.tmail.combined.identity;
+
+import java.util.Iterator;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.Username;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.user.api.model.User;
+import org.apache.james.user.cassandra.CassandraUsersDAO;
+import org.apache.james.user.ldap.ReadOnlyUsersLDAPRepository;
+import org.apache.james.user.lib.UsersRepositoryImpl;
+
+public class CombinedUsersRepository implements UsersRepository {
+
+    private final ReadOnlyUsersLDAPRepository readOnlyUsersLDAPRepository;
+    private final UsersRepositoryImpl<CassandraUsersDAO> cassandraUsersRepository;
+
+    @Inject
+    public CombinedUsersRepository(ReadOnlyUsersLDAPRepository readOnlyUsersLDAPRepository,
+                                   UsersRepositoryImpl<CassandraUsersDAO> cassandraUsersRepository) {
+        this.readOnlyUsersLDAPRepository = readOnlyUsersLDAPRepository;
+        this.cassandraUsersRepository = cassandraUsersRepository;
+    }
+
+    @Override
+    public void addUser(Username username, String password) throws UsersRepositoryException {
+        throw new UsersRepositoryException("addUser method is unsupported");
```

Comment:
IMO we should add the user to Cassandra IF he exists in the LDAP.

This will be useful  for later synchronizations with LSC.

Reply:
First, I thought we will generate random passwords, and this logic is processed in the service layer. 
But I have another approach now

Reply:
Which is? :)

Reply:
Like as Benoit said "we should add the user to Cassandra IF he exists in the LDAP" :D

Reply:
That does not answer about the password. I don't think you should store the password to 2 different places, even more if the one in Cassandra is not used. In terms of security, it's not great IMO

Reply:
We don't save "real" passwords in Cassandra (24 randomly generated characters). 
Because the test user method still calls directly to LDAP to verify. 

I think keeping the current table help us low coding cost. (save password but don't use it)